index.js

```diff
@@ -42,11 +46,13 @@ exports.kinesisHandler = (records, opts = {}, context) => {
       apiHelper.getOAuthToken(CACHE['access_token']),
       streamsClient.decodeData(schema, records.map(i => i.kinesis.data))
     ]).then(result => {
-      // Both promises were fulfilled
+      // The access_token has been obtained and all records have been grouped by source
+      // Next, we need create a string from the record and nyplSource keys to perform a GET request
+      // to the ItemService
       CACHE['access_token'] = result[0];
-      const decodedKinesisData = result[1];
-
-      console.log(decodedKinesisData, CACHE);
+      const groupedRecordsBySource = apiHelper.groupRecordsBy(result[1], 'nyplSource');
```

Comment:
I understand the need to group the records by `nyplSource` but would it also be good to save/keep the array of records (i.e. `result[0]`) since that's really the most important aspect as it's the array of `HoldRequests`? Just thinking through how you might need to refer to these records later.